### PR TITLE
Icon: Update iOS and Android specs

### PR DESCRIPTION
**Pull request by hectoid**
#### What changed?

Added specs to iOS docs. Android: Fixed typo on naming, added touch target spec, updated dimensions to 24px.

#### Why?

To fix mistakes made on Android and add clarity to Touch target dimensions. To add specs to iOS for development

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-5725)
- [TDD](https://coda.io/d/Gestalt-Team-Folder_d2LeXkQ1kVX/Icon_su4lE#_luqH9)
- [Figma](https://www.figma.com/file/AHcKJDgb7E7YswlgW1wY8E/Gestalt-for-iOS?node-id=19236%3A74589&t=D3hNy73JBEb5S9mN-1)

### Checklist

- [x] Added unit and Flow Tests
- [x] Added documentation + accessibility tests
- [x] Checked dark mode, responsiveness, and right-to-left support
- [x] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/markdown/ios/icon.md [hunk 1]
```diff
@@ -6,30 +6,34 @@ fullwidth: true
 
 <ImgContainer src="https://i.pinimg.com/originals/1d/52/31/1d5231c99699e72628135de682fede30.jpg" alt="Five example icons in a row. Icons include share, check, back, filter, and menu." />
 
+## Icon assets
+
+Download svg assets from our [iOS icon GitHub repo](#).
```
Comment (AlbertCarreras) [inline docs/markdown/ios/icon.md:11]: Should we have a dead link?

**Comment by AlbertCarreras** [comment]
@hectoid look here to find problem https://github.com/pinterest/gestalt/actions/runs/4431559954/jobs/7774646862

https://capture.dropbox.com/lHFDFf3quzro9Bup

**Review by AlbertCarreras — APPROVED** [review]
LGTM, just the test failing

File: docs/markdown/ios/icon.md (hunk 1 above)
Comment (hectoid) [reply docs/markdown/ios/icon.md]: no. good catch. i removed the link

**Pull request merged**